docs/tutorial.rst

```diff
@@ -433,6 +433,31 @@ The *context* variable in all cases is an instance of
 
 .. _docid.tutorial.environmental-controls:
 
+Using Context data in steps
+---------------------------
+
+Like shown above, you can use the data in "context" to pass data between steps.
+Variables stored there can also be used in a scenario like so:
+
+.. code-block:: python
+
+    @given('I request a new widget for an account via SOAP')
+    def step_impl(context):
+        client = Client("http://127.0.0.1:8000/soap/")
+        response = client.Allocate(first='Firstname', last='Lastname')
+        context.widget_id = response['id']
+
+    @then('I delete the widget with ID "{id}"')
+    def step_impl(context, id):
+        client = Client("http://127.0.0.1:8000/soap/")
+        client.delete(id)
+
+.. code-block:: gherkin
+
+  Scenario: request and delete widget
+    Given I request a new widget for an account via SOAP
+    When I delete the widget with ID "<widget_id>"
```

Comment:
Interesting! Is this `"<syntax>"` officially supported by the Gherkin specification?

Reply:
As a second thought, why would we want to mix the description of the behavior and details of the implementation? Shouldn't we better discourage such a style of programming?

Reply:
Our main use case is to enable highly reusable step implementations, shared among many feature files. Specifying an ID here is rather bad form, I agree!

To be honest, I had quite a bit of trouble figuring out an example. Here is how we use it:

```
Given there is an uberspace account
When I input "uberspace setup mail"
Then the directory "/home/<username>/users" does exist
```

We don't know the account name beforehand. So the `@given(u'there is an uberspace account')` step sets `context.username` to be used by the other steps.

Maybe we can work out a better example starting from this?

Reply:
The "normal way" to avoid this is to use a "persona" oriented approach (or user-roles/host-roles and instance numbers, if possible/sensible).You basically use names to identify configuration dataset that you want to use.
The dataset can be provided by using the userdata mechanism (or similar other mechanisms). Within the step you just need to identify which dataset to use and the step specifies what should be done (note: The identification of the dataset may occur in an earlier step, often a given-step).

HINT:
The concrete IPv4 address may be unnecessary technical details in BDD steps (a smell that hints that you may provide too many low-level details which may clutter your steps and hide what is really important in the steps).

RELATED: Personas
* https://www.interaction-design.org/literature/article/personas-why-and-how-you-should-use-them
* https://www.testingtime.com/en/blog/complete-guide-how-to-create-personas-based-on-data/

Reply:
> Interesting! Is this `"<syntax>"` officially supported by the Gherkin specification?

It is the placeholder syntax in ScenarioOutline templates.
That problem is that newer Gherkin version want to make Scenario and ScenarioOutline interchangeable.
Therefore, you can use the keyword `Scenario` but it is actually a `ScenarioOutline` / `ScenarioTemplate`.
This suggested placeholder syntax above will prevent to support this Gherkin functionality in `behave` in the future (same syntax used for two different things).